cachi2/core/models/sbom.py

```diff
@@ -93,3 +97,290 @@ class Sbom(pydantic.BaseModel):
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_spdx(self) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM."""
+        packages = []
+        relationships = []
+
+        packages.append(
+            SPDXPackage(
+                name="",
+                versionInfo="",
+                SPDXID="SPDXRef-DocumentRoot-File-",
+            )
+        )
+
+        for component in self.components:
+            annotations = []
+            for prop in component.properties:
+                annotations.append(
+                    SPDXPackageAnnotation(
+                        annotator="cachi2",
+                        annotationDate=datetime.datetime.now().isoformat(),
+                        annotationType="OTHER",
+                        comment=json.dumps({"name": f"{prop.name}", "value": f"{prop.value}"}),
+                    )
+                )
+            package_hash = SPDXPackage._calculate_package_hash_from_dict(
+                {
+                    "name": component.name,
+                    "version": component.version,
+                    "purl": component.purl,
+                }
+            )
+
+            packages.append(
+                SPDXPackage(
+                    SPDXID=f"SPDXID-Package-{component.name}-{component.version}-{package_hash}",
```

Comment:
Golang's stlib modules don't have a version specified in the SBOM. This will make the SPDXID end up like:
```
SPDXID-Package-bufio-None-50fdaec28e33e601d7d652bcd07106e339cb553fac4be5ff9d34de2d4cef23d8
```

Not sure if having the `None` there is problem, though. It would be slightly better to just omit it, IMO.